Engine requires bourbon, fixes #615

kpheasey opened the pull request.

JoelQ commented:
I'd love to get some testing around this. Currently the test suite is green (if you rebase the latest `master`) without this change. Can we make the `sample_app` fail the same way as everyone else and then this fix makes it go green?

kpheasey commented on .gitignore:
```diff
@@ -14,3 +14,4 @@
 /tags
 /tmp/*
 /spec/example_app/tmp/*
+.idea
```
Never knew there was global .gitignore, thanks for the knowledge!

I'll revert it

kpheasey commented:
Rebased with master and removed the .gitignore stuff.

Not sure how to test the require statement in the sample app.  Since the sample app uses the engine's gemfile bourbon gets included.  When bourbon is not in the gemfile, it is not required by the engine.

JoelQ commented:
Ah, I see what you're saying.

I think we can add `require: false` to bourbon in the `Gemfile`. This will prevent it from being auto-required. Doing this on my machine caused all the feature specs to fail with:

```
ActionView::Template::Error:
        File to import not found or unreadable: bourbon.
```

Adding in your require to `engine.rb` then caused the test suite to go green.

Alternatively, we could try to go down the path of having a separate `Gemfile` for the sample app.

Thoughts?

cc @danbee @BenMorganIO

kpheasey commented:
I believe dependencies should be defined in the gemspec, http://guides.rubyonrails.org/engines.html#other-gem-dependencies <http://guides.rubyonrails.org/engines.html#other-gem-dependencies>

If you define a dependency in the gemspec, it will need to be explicitly required upon initialization.  This would avoid further confusion and not require testing.
> On Oct 28, 2016, at 3:35 PM, Joël Quenneville <notifications@github.com> wrote:
> 
> Ah, I see what you're saying.
> 
> I think we can add require: false to bourbon in the Gemfile. This will prevent it from being auto-required. Doing this on my machine caused all the feature specs to fail with:
> 
> ActionView::Template::Error:
>         File to import not found or unreadable: bourbon.
> Adding in your require to engine.rb then caused the test suite to go green.
> 
> Alternatively, we could try to go down the path of having a separate Gemfile for the sample app.
> 
> Thoughts?
>

BenMorganIO commented:
@kpheasey could you squash your three commits? I think they could all be just one in this instance.

JoelQ commented:
> I believe dependencies should be defined in the gemspec

Good point. It looks like we have some dependencies just in the Gemfile, some just in the gemspec, and some in both. Not sure why that happened.

Do you want turn this PR into a migration from using the Gemfile to using the gemspec? Doing that should both fix the bourbon problem and fail the specs when not requiring he proper files.

kpheasey commented:
I've completely screwed this up trying to squash, I'm going to open a new pull request with a separate branch, sorry about that.

JoelQ commented:
No worries @kpheasey 😃
